File: src/IMG/IMG_Item.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package IMG;

/**
 *
 * @author Kilian
 */
public class IMG_Item {
    private int type;
    private int offset;
    private int size;
    private String name;

    private boolean resource = false;
    private int flags = 0;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getOffset() {
        return offset;
    }

    public void setOffset(int offset) {
        this.offset = offset;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        if(type < 1000){
            resource = true;
        }
        this.type = type;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public int getFlags() {
        return flags;
    }

    public void setFlags(int flags) {
        this.flags = flags;
    }

    public boolean isResource() {
        return resource;
    }

    public void setResource(boolean resource) {
        this.resource = resource;
    }



}

File: src/main/java/nl/shadowlink/tools/shadowlib/img/ImgItem.kt
package nl.shadowlink.tools.shadowlib.img

/**
 * @author Shadow-Link
 */
class ImgItem {

    var type: Int = 0
        set(type) {
            if (type < 1000) {
                isResource = true
            }
            field = type
        }

    var offset: Int = 0
    var size: Int = 0
    var name: String? = null

    var isResource = false
    var flags = 0
}
